spec.md

```diff
@@ -286,6 +286,59 @@ circuit Foo :
   public module Foo enablelayer A :
 ```
 
+## Formal Tests
+
+Formal tests are names top-level constructs which allow for test-harnesses to be definied 
+as bounded model checking problems. Formal tests have 3 operands: 
+- A symbol that the test can be referred to with.  
+- The name of the test harness module.  
+- A bound for the bounded model checking problem, which refers to the number of cycles that 
```

Comment:
what is a 'cycle'? Is there an assumed clock for a formal test?

Reply:
Formal tests are converted into state-transition systems, so I guess 'time-step' is a better term. The clock used for the time step is whatever clocks are tied to the assertions or stateful elements used through the harness, so a bound of 20 defines a state transition system that is 20 nodes deep.